jena-core/src/main/java/org/apache/jena/vocabulary/OWL.java

```diff
@@ -20,178 +20,11 @@
-        public static Property inverseOf()                  { return property( "inverseOf" ); }
-        public static Property backwardCompatibleWith()     { return property( "backwardCompatibleWith" ); }
-        public static Property differentFrom()              { return property( "differentFrom" ); }
-        public static Property priorVersion()               { return property( "priorVersion" ); }
-        public static Property imports()                    { return property( "imports" ); }
-        public static Property allValuesFrom()              { return property( "allValuesFrom" ); }
-        public static Property unionOf()                    { return property( "unionOf" ); }
-        public static Property hasValue()                   { return property( "hasValue" ); }
-        public static Property someValuesFrom()             { return property( "someValuesFrom" ); }
-        public static Property disjointWith()               { return property( "disjointWith" ); }
-        public static Property cardinality()                { return property( "cardinality" ); }
-        public static Property intersectionOf()             { return property( "intersectionOf" ); }
-        public static Resource Thing()                      { return resource( "Thing" ); }
-        public static Resource DataRange()                  { return resource( "DataRange" ); }
-        public static Resource Ontology()                   { return resource( "Ontology" ); }
-        public static Resource DeprecatedClass()            { return resource( "DeprecatedClass" ); }
-        public static Resource AllDifferent()               { return resource( "AllDifferent" ); }
-        public static Resource DatatypeProperty()           { return resource( "DatatypeProperty" ); }
-        public static Resource SymmetricProperty()          { return resource( "SymmetricProperty" ); }
-        public static Resource TransitiveProperty()         { return resource( "TransitiveProperty" ); }
-        public static Resource DeprecatedProperty()         { return resource( "DeprecatedProperty" ); }
-        public static Resource AnnotationProperty()         { return resource( "AnnotationProperty" ); }
-        public static Resource Restriction()                { return resource( "Restriction" ); }
-        public static Resource Class()                      { return resource( "Class" ); }
-        public static Resource OntologyProperty()           { return resource( "OntologyProperty" ); }
-        public static Resource ObjectProperty()             { return resource( "ObjectProperty" ); }
-        public static Resource FunctionalProperty()         { return resource( "FunctionalProperty" ); }
-        public static Resource InverseFunctionalProperty()  { return resource( "InverseFunctionalProperty" ); }
-        public static Resource Nothing()                    { return resource( "Nothing" ); }
-    }
+@Deprecated
+public class OWL extends OWL1 {
```

Comment:
This cases 424 warnings on the Jena code base!

200 are in jena-core:src/main/java:org.apache.jena.ontology 
50 are in  jena-core:src/main/java:org.apache.jena.reasoner

155 are in jena-core tests

Shouldn't these all be OWL1?

As vocabularies, is there anything OWL2 that conflicts with OWL1?

Would it work if we change all uses in Jena to be OWL1 where it must be OWL1 (legacy ontology, reasoner) then have `public class OWL extends OWL2`?

Reply:
there are 4 constants in OWL1 that are not in OWL2: `NAMESPACE`, `LITE_LANG`, `DL_LANG`, `FULL_LANG`
we can do like this:
```java
class OWL {
        public static final Resource NAMESPACE = ResourceFactory.createResource(NS);
        public static final Resource OWL1_FULL_LANG = ResourceFactory.createResource(getURI());
        public static final Resource OWL1_DL_LANG = ResourceFactory.createResource("http://www.w3.org/TR/owl-features/#term_OWLDL");
        public static final Resource OWL1_LITE_LANG = ResourceFactory.createResource("http://www.w3.org/TR/owl-features/#term_OWLLite");
        @Deprecated
        public static final Resource FULL_LANG = OWL1_FULL_LANG;
        @Deprecated
        public static final Resource DL_LANG = OWL1_DL_LANG;
        @Deprecated
        public static final Resource LITE_LANG = OWL1_LITE_LANG;
...        
```
And make OWL2 deprecated to avoid confusion.
in this case we do not need OWL1 vocabulary at all

Reply:
With this new initialization I observe error when run particular test through IDE:
```
java.lang.ExceptionInInitializerError
	at org.apache.jena.riot.RDFParserRegistry.initStandard(RDFParserRegistry.java:74)
	at org.apache.jena.riot.RDFParserRegistry.init(RDFParserRegistry.java:57)
	at org.apache.jena.riot.RDFParserRegistry.<clinit>(RDFParserRegistry.java:52)
	at org.apache.jena.riot.RIOT.init(RIOT.java:83)
	at org.apache.jena.riot.system.InitRIOT.start(InitRIOT.java:29)
	at org.apache.jena.base.module.Subsystem.lambda$initialize$1(Subsystem.java:117)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.apache.jena.base.module.Subsystem.forEach(Subsystem.java:193)
	at org.apache.jena.base.module.Subsystem.forEach(Subsystem.java:169)
	at org.apache.jena.base.module.Subsystem.initialize(Subsystem.java:115)
	at org.apache.jena.sys.JenaSystem.init(JenaSystem.java:89)
	at org.apache.jena.graph.NodeFactory.<clinit>(NodeFactory.java:39)
	at org.apache.jena.rdf.model.impl.ResourceImpl.fresh(ResourceImpl.java:150)
	at org.apache.jena.rdf.model.impl.ResourceImpl.<init>(ResourceImpl.java:86)
	at org.apache.jena.rdf.model.ResourceFactory$Impl.createResource(ResourceFactory.java:308)
	at org.apache.jena.rdf.model.ResourceFactory.createResource(ResourceFactory.java:94)
	at org.apache.jena.vocabulary.RDF.resource(RDF.java:54)
	at org.apache.jena.vocabulary.RDF.<clinit>(RDF.java:65)
	at org.apache.jena.vocabulary.RDF$Nodes.<clinit>(RDF.java:193)
	at org.apache.jena.ontapi.utils.Graphs.listOntologyNodes(Graphs.java:659)
	at org.apache.jena.ontapi.utils.Graphs.ontologyNode(Graphs.java:626)
	at org.apache.jena.ontapi.utils.Graphs.ontologyNode(Graphs.java:613)
	at org.apache.jena.ontapi.utils.Graphs.findOntologyNameNode(Graphs.java:561)
	at org.apache.jena.ontapi.utils.Graphs.findOntologyNameNode(Graphs.java:544)
	at org.apache.jena.ontapi.impl.repositories.PersistentGraphRepository.lambda$baseOntGraphs$2(PersistentGraphRepository.java:64)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1707)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.apache.jena.ontapi.impl.repositories.PersistentGraphRepository.collectUnions(PersistentGraphRepository.java:53)
	at org.apache.jena.ontapi.impl.repositories.PersistentGraphRepository.<init>(PersistentGraphRepository.java:49)
	at org.apache.jena.ontapi.PersistentGraphRepositoryTest.testSeveralRepositories(PersistentGraphRepositoryTest.java:66)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.jena.rdf.model.Resource.getURI()" because "org.apache.jena.vocabulary.RDF.xmlLiteral" is null
	at org.apache.jena.riot.lang.ReaderTriX.<clinit>(ReaderTriX.java:119)
	... 38 more
```

It can be fixed by removing `static { JenaSystem.init(); }` in `NodeFactory`:
![image](https://github.com/apache/jena/assets/17473366/e8e4e622-9a97-437a-89e9-b23660a4ec61)

So should I remove this line or return back that sophisticated initialization?

Reply:
I don't see `PersistentGraphRepositoryTest` in the PR code base - I have this pulled PR as of commit 648852b. The warnings count looks the same as before. 

1/
The situation in the stacktrace usually means the code is already inside `JenaSystem.init`.

There is call to `JenaSystem.init` here because code may use `NodeFactory` as a first call into Jena.
tests suite can sometimes need their own `JenaSystem.init` because they do tend to go straight the non-public code they are testing.

`PersistentGraphRepository` looks like a entry point (a place where a first call happens) - maybe it needs `static { `JenaSystem.init(): }`

Calling `JenaSystem.init` should always be safe. I can't see from at the stacktrace how the code got to `org.apache.jena.riot.lang.ReaderTriX.<clinit>(ReaderTriX.java:119)`.

`PersistentGraphRepositoryTest` may need a call to `JenaSystem.init`. If that does not help, then, yes, this may need to go back to the old way of initializing. which was for this this case but IIRC before ServiceLoader came along.

2/
This isn't the cause  - it may help generally.

jena-ontapi does not have a `src/main/resources/META-INF/services/org.apache.jena.sys.JenaSubsystemLifecycle` file. 

See `jena-core` for an example. Having one of these, controls the initlialization order in Jena so it can go after jena-core before jena-arq at level 15 (see `JenaSystem` for details).

Reply:
PersistentGraphRepositoryTest is here https://github.com/apache/jena/pull/2496 - it is another independent PR

Can you please give me an example of warning. This is strange, since in core and arq there are only renaming.

Reply:
Added `src/main/resources/META-INF/services/org.apache.jena.sys.JenaSubsystemLifecycle` with content:
```
org.apache.jena.riot.system.InitRIOT
org.apache.jena.sparql.system.InitARQ
org.apache.jena.rdfs.sys.InitRDFS
```

![image](https://github.com/apache/jena/assets/17473366/916a2bff-79d1-4368-a96c-112b2b345691)